src/app/components/Filters/FieldsetDate.jsx

```diff
@@ -1,27 +1,99 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import Actions from '../../actions/Actions.js';
+import {
+  extend as _extend,
+} from 'underscore';
+import NumberFormat from 'react-number-format';
+
+class FieldsetDate extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      dateAfter: '0',
+      dateBefore: '0',
+      selectedFacets: this.props.selectedFacets,
+    };
+
+    this.inputChange = this.inputChange.bind(this);
+  }
+
+  /**
+   * inputChange()
+   * Updates the states and the store based on the input values.
+   *
+   */
+  inputChange(e) {
+    const value = e.target.value;
+    let displayValue = '';
+
+    if (e.target.name === 'start-date') {
+      displayValue = 'dateAfter';
+    } else if (e.target.name === 'end-date') {
+      displayValue = 'dateBefore';
+    }
+
+    if (displayValue) {
+      this.setState(
+        {
+          [displayValue]: value,
+          selectedFacets: _extend(
+            this.state.selectedFacets,
+            { [displayValue]: value }
+          ),
+        },
+        () => {
+          Actions.updateSelectedFacets(this.state.selectedFacets);
+        }
+      );
+    }
+  }
 
-class FilterPopup extends React.Component {
   render() {
+    let errorMessage = '';
+
+    if (this.state.dateAfter && this.state.dateBefore) {
+      if (Number(this.state.dateBefore) < Number(this.state.dateAfter)) {
+        errorMessage = 'end year should be later than start year.';
+      }
+    } else {
+      errorMessage = '';
+    }
+
     return (
       <fieldset>
         <legend>Date</legend>
-        <div>
+        <div id="input-container">
           <label htmlFor="start-date">Start Year
-            <input id="start-date" name="start-date" className="form-text" type="date" />
+            <NumberFormat
+              id="start-date"
+              name="start-date"
+              className="form-text"
+              onChange={this.inputChange}
+              format="####"
+            />
           </label>
           <label htmlFor="end-date">End Year
-            <input id="end-date" type="date" className="form-text" name="end-date" />
+            <NumberFormat
+              id="end-date"
+              name="end-date"
+              className="form-text"
+              onChange={this.inputChange}
+              format="####"
+            />
           </label>
           <span>The Start year cannot be later than the end year</span>
+          <br />
+          <span id="error-message">{errorMessage}</span>
```

Comment:
This isn't the right structure for an inline error message. For example, it should have an `aria-live` element and the field label should be updated to be the formal field label (either `start` or `end` year) and the error message. You can the reference GALC form for the full pattern.

Reply:
I can update the DOM. But the error message serves more like a placeholder now. Do we know how to present the error when the user types in incorrect dates?